src/service/VerifyService.ts

```diff
@@ -41,91 +40,100 @@ const defaultExpectedVersions: ExpectedVersions = {
-        report.lines.forEach((line) => {
-            if (line.recommendation) {
-                this.logger.error(`${line.header}  - Error! - ${line.message} - ${line.recommendation}`);
-            } else {
-                this.logger.info(`${line.header} - OK! - ${line.message}`);
-            }
-        });
-    }
-
-    public validateReport(report: VerifyReport): void {
-        const errors = report.lines.filter((r) => r.recommendation);
-        if (errors.length) {
-            throw new Error(
-                'There has been an error. Check the report: \n' +
-                    errors.map((line) => ` - ${line.header}  - Error! - ${line.message} - ${line.recommendation}`).join('\n'),
-            );
-        }
-    }
+export interface VerifyAction {
+    shouldRun(lines: ReportLine[]): Promise<boolean>;
+    verify(): Promise<ReportLine>;
+}
 
+export class AppVersionService {
+    public static readonly semverOptions = { loose: true };
+    constructor(private readonly logger: Logger) {}
     public loadVersion(text: string): string | undefined {
         return text
             .replace(',', '')
             .split(' ')
             .map((word) => {
-                const coerce = semver.coerce(word.trim(), this.semverOptions);
+                const coerce = semver.coerce(word.trim(), AppVersionService.semverOptions);
                 return coerce?.raw;
             })
             .find((a) => a)
             ?.trim();
     }
 
-    public async testNodeJs(expectedVersion: string): Promise<ReportLine> {
-        const header = 'NodeVersion';
-        const recommendationUrl = `https://nodejs.org/en/download/package-manager/`;
-        const output = process.versions.node;
-        return this.verifyInstalledApp(async () => output, header, expectedVersion, recommendationUrl);
+    public async loadVersionFromCommand(command: string): Promise<string | undefined> {
+        return this.loadVersion((await BootstrapUtils.exec(this.logger, command)).stdout.trim());
     }
 
-    public async testDocker(expectedVersion: string): Promise<ReportLine> {
-        const header = 'Docker Version';
-        const command = 'docker --version';
-        const recommendationUrl = `https://docs.docker.com/get-docker/`;
-        return this.verifyInstalledApp(() => this.loadVersionFromCommand(command), header, expectedVersion, recommendationUrl);
+    public async verifyInstalledApp(
+        versionLoader: () => Promise<string | undefined>,
+        header: string,
+        minVersion: string,
+        recommendationUrl: string,
+    ): Promise<ReportLine> {
+        try {
+            const version = await versionLoader();
+            if (!version) {
+                return {
+                    header,
+                    message: `Version could not be found! Output: ${versionLoader}`,
+                    recommendation: `At least version ${minVersion} is required. Check ${recommendationUrl}`,
```

Comment:
`At least version ${minVersion} is required. Check ${recommendationUrl}` is used in all recommendations, except in middle it has a prefix with current version status.

Reply:
Refactored common string